hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsModule.java

```diff
@@ -253,4 +259,36 @@ static Supplier<CallLocalEvmTxProcessor> provideCallLocalEvmTxProcessorFactory(
         return () -> new CallLocalEvmTxProcessor(
                 codeCache, livePricesSource, dynamicProperties, gasCalculator, mcps, ccps, aliasManager);
     }
+
+    @Provides
+    @Singleton
+    @Named("HederaPrecompiledContractsDetector")
+    static Predicate<Address> provideHederaPrecompiledContractsDetector() {
+        // all addresses between 0-750 (inclusive) are treated as system accounts
+        // from the perspective of the EVM when executing Call, Balance, and SelfDestruct operations
+        return address -> address.numberOfLeadingZeroBytes() >= 18
+                && (Integer.compareUnsigned(address.getInt(16), 359) == 0
+                        || Integer.compareUnsigned(address.getInt(16), 360) == 0
+                        || Integer.compareUnsigned(address.getInt(16), 361) == 0);
+    }
+
+    @Provides
+    @Singleton
+    @Named("HederaSystemAccountDetector")
+    static Predicate<Address> provideHederaSystemAccountDetector() {
+        // all addresses between 0-750 (inclusive) are treated as system accounts
+        // from the perspective of the EVM when executing Call, Balance, and SelfDestruct operations
+        return address ->
+                address.numberOfLeadingZeroBytes() >= 18 && Integer.compareUnsigned(address.getInt(16), 750) <= 0;
```

Comment:
This `750` should probably be rendered as [`HederaNumbers.NUM_RESERVED_SYSTEM_ENTITIES`](https://github.com/hashgraph/hedera-services/pull/6373) here.

Reply:
I'll fix in a subsequent PR (this one's been merged) and look for other similar things too.